src/components/heatmap/heatmap.tsx

```diff
@@ -0,0 +1,344 @@
+     * Return an object with all the required layout parameters.
+     * @example
+     * {
+     * 	height: height of the heatmap,
+     * 	width: width of the heatmap,
+     * 	margin: {
+     * 		t: top margin of the heatmap,
+     * 		r: right margin of the heatmap,
+     * 		b: bottom margin of the heatmap,
+     * 		l: left of the heatmap
+     * 	},
+     * 	xTickAngle: inclination of x axis labels,
+     *  yTickAngle: inclination of y axis labels,
+     * 	tickFont: font to be used in labels
+     * }
+     */
+  const getHeatmapLayoutParams = (input: any, longestXTick: number, longestYTick: number, lengthY: number) => {
+    let height;
+    let yTickAngle;
+    const tMargin = 25;
+    let rMargin, bMargin, lMargin;
+
+    if (longestXTick <= 18) {
+      height = longestXTick * 9 + lengthY * 10 + 50;
+      bMargin = 8.4 * longestXTick;
+    } else if (longestXTick <= 22) {
+      height = longestXTick * 9 + lengthY * 10 + 55;
+      bMargin = 8.4 * longestXTick;
+    } else if (longestXTick <= 30) {
+      height = longestXTick * 8.8 + lengthY * 10 + 55;
+      bMargin = 8.2 * longestXTick;
+    } else {
+      height = longestXTick * 8.8 + lengthY * 10 + 45;
+      bMargin = 8 * longestXTick;
+    }
+
+    if (lengthY === 1) {
+      yTickAngle = -90;
+      lMargin = 30;
+      rMargin = 20;
+    } else {
+      yTickAngle = 0;
+      lMargin = 20 + longestYTick * 7;
+      rMargin = 5;
+    }
+
+    const layoutParams = {
+      height: height,
+      width: input.width,
+      margin: {
+        t: tMargin,
+        r: rMargin,
+        b: bMargin,
+        l: lMargin,
+      },
+      xTickAngle: input.xTickAngle,
+      yTickAngle: yTickAngle,
+      tickFont: input.tickFont
+    };
+    return layoutParams;
+  }
+
+
+  /** It can have following input parameters in the heatmap-config.js file in presenattion object:
+   * @param {any} plot: heatmap object
+   * Return data and layout object that can be passed to plotly component to generate heatmap
+   * @example:
+   * Plot
+   * id: id of heatmap
+   * rows{  
+   * colorbar
+   * type: type of plot (heatmap)
+   * x: data to be shown on X axis
+   * y: data to be shown on y axis
+   * z: z values (affect the color of datapoint)
+   * title: title of heatmap
+   * }
+   * ----------
+   * Layout: 
+   * height
+   * margin: {t, r, b, }
+   * title
+   * width
+   * xaxis: {tickangle, tickfont, tickvals, ticktext}
+   * yaxis: {tickangle, tickvals, ticktext, tickfont} 
+   */
+
+  const createPlotLayoutParams = (plot: any = {}) => {
+    const configPresentation = Object.assign({}, config.presentation);
+    let layout = {};
+    if (plot) {
+      const longestXTick = plot?.rows?.x.reduce((a: any, b: any) => { return a?.length > b?.length ? a : b; });
+      const longestYTick = plot?.rows?.y.reduce((a: any, b: any) => { return a?.length > b?.length ? a : b; });
+      const inputParams = {
```

Comment:
This needs a type defined for this object. Since we are creating the object ourself, we can define each property we know will be present and their types.